src/main/java/com/virtudoc/web/entity/UserAccount.java

```diff
@@ -49,6 +74,28 @@ public UserAccount(String username, String password, String role) {
         this.role = role;
     }
 
+    /**
+     * Constructor for UserAccount class. Utilizes the NewUserDTO to ensure that protected fields
+     * such as ROLE are not exposed in the controller.
+     * @param userDTO DTO object from frontend Model.
+     * @throws Exception Given password pair does not match.
+     * @see NewUserDTO
+     */
+    public UserAccount(NewUserDTO userDTO) throws Exception {
+        if (userDTO.getPassword().equals(userDTO.getConfirmedPassword())) {
```

Comment:
Sorry, this is a bug from my code. This conditional needs a `!` in front since we only want to throw an error if they don't match. In the interest of getting this PR out the door, this could be fixed in another PR later since it's not related to your new code.